ui/components/app/nft-details/nft-details.tsx

```diff
@@ -0,0 +1,861 @@
+            ariaLabel={t('back')}
+            iconName={IconName.ArrowLeft}
+            onClick={() => history.push(DEFAULT_ROUTE)}
+            data-testid="nft__back"
+          />
+          <NftOptions
+            onViewOnOpensea={
+              openSeaLink
+                ? () => global.platform.openTab({ url: openSeaLink })
+                : null
+            }
+            onRemove={onRemove}
+          />
+        </Box>
+        <Box
+          display={Display.Flex}
+          justifyContent={JustifyContent.center}
+          marginBottom={8}
+          marginTop={1}
+        >
+          <Box className="nft-details__nft-item">
+            <NftItem
+              src={isImageHosted ? image : nftImageURL}
+              alt={image ? nftImageAlt : ''}
+              name={name}
+              tokenId={tokenId}
+              networkName={currentChain.nickname}
+              networkSrc={currentChain.rpcPrefs?.imageUrl}
+              isIpfsURL={isIpfsURL}
+              onClick={handleImageClick}
+              clickable
+            />
+          </Box>
+        </Box>
+        <Box>
+          {name || collection?.name ? (
+            <Box display={Display.Flex} alignItems={AlignItems.center}>
+              <Text
+                variant={TextVariant.headingMd}
+                fontWeight={FontWeight.Bold}
+                color={TextColor.textDefault}
+                fontStyle={FontStyle.Normal}
+                style={{ fontSize: '24px' }}
+                data-testid="nft-details__name"
+              >
+                {name || collection?.name}
+              </Text>
+              {collection?.openseaVerificationStatus === 'verified' ? (
+                <Icon
+                  marginLeft={1}
+                  name={IconName.SecurityTick}
+                  color={IconColor.primaryDefault}
+                />
+              ) : null}
+            </Box>
+          ) : null}
+
+          <NftDetailDescription value={description} />
+
+          <Box
+            marginTop={4}
+            marginBottom={4}
+            display={Display.Flex}
+            gap={4}
+            flexWrap={FlexWrap.Wrap}
+          >
+            {hasLastSalePrice || hasFloorAskPrice ? (
+              <>
+                <NftDetailInformationFrame
+                  frameClassname="nft-details__nft-frame"
+                  title={t('boughtFor')}
+                  frameTextTitleProps={{
+                    textAlign: TextAlign.Center,
+                    color: TextColor.textAlternative,
+                    variant: TextVariant.bodyMdMedium,
+                  }}
+                  frameTextTitleStyle={{
+                    fontSize: '10px',
+                    lineHeight: '16px',
+                  }}
+                  value={
+                    lastSale?.price?.amount?.usd
+                      ? formatCurrency(
+                          `${lastSale?.price?.amount?.usd}`,
+                          currency,
```

Comment:
Does this work if the user configures a currency other than USD in MM settings?

`lastSale?.price?.amount?.usd` looks like it would always be USD, but then would this format it into euros without converting?